trainer.py

```diff
@@ -92,12 +101,52 @@ def get_epochs(self):
     def get_test_data(self):
         return self._x_test, self._y_test
 
+    def get_prune_threshold(self):
+        return self._prune_threshold
+
     def add_checkpoint_callback(self, callback):
         self._checkpoint_callbacks.append(callback)
 
-    def go(self):
+    def _train(self):
         raise NotImplementedError
 
+    def _prune(self, model):
+        """ Prunes model (in place) using prune_threshold """
+        BaseTrainer.prune_trained_model(model, self._prune_threshold)
+
+    @staticmethod
+    def prune_trained_model(model, threshold):
+        if math.isnan(threshold):
+            return
+        # Layer 0 has no input edges, start from layer 1
+        pruned_edges = 0
+        for i in range(1, len(model.layers)):
```

Comment:
use use get_weighted_layers_indices to get the specific layers that have weights to get the list of layers that have weights.